File: sample.ts
// sample.ts — a small TypeScript example for meer's preview demo

type Repository = {
  owner: string
  name: string
  isPrivate: boolean
  defaultBranch: string
}

type Branch = {
  name: string
  commit: string
}

/**
 * A friendly helper that formats a repository's full name
 * with an optional visibility marker.
 */
export function formatRepoName(repo: Repository): string {
  const visibility = repo.isPrivate ? ' (private)' : ''
  return `${repo.owner}/${repo.name}${visibility}`
}

/**
 * Find the default branch from a list, falling back to "main".
 */
export function pickDefaultBranch(
  branches: Branch[],
  hint?: string,
): Branch | undefined {
  if (hint) {
    const found = branches.find((b) => b.name === hint)
    if (found) return found
  }
  return (
    branches.find((b) => b.name === 'main') ??
    branches.find((b) => b.name === 'master') ??
    branches[0]
  )
}

// Example usage
const repos: Repository[] = [
  { owner: 'misstouch-taro', name: 'meer', isPrivate: false, defaultBranch: 'main' },
  { owner: 'misstouch-taro', name: 'meer-demo', isPrivate: false, defaultBranch: 'main' },
]

for (const repo of repos) {
  console.log(formatRepoName(repo))
}

File: sample.py
"""sample.py — a tiny Python example for meer's preview demo."""

from dataclasses import dataclass
from typing import Iterable


@dataclass
class Repository:
    """A minimal representation of a code repository."""

    owner: str
    name: str
    is_private: bool = False
    default_branch: str = "main"

    @property
    def full_name(self) -> str:
        suffix = " (private)" if self.is_private else ""
        return f"{self.owner}/{self.name}{suffix}"


def list_public_repos(repos: Iterable[Repository]) -> list[Repository]:
    """Return only the public repositories."""
    return [r for r in repos if not r.is_private]


def greet(repo: Repository) -> str:
    return f"Hello from {repo.full_name}!"


if __name__ == "__main__":
    samples = [
        Repository("misstouch-taro", "meer"),
        Repository("misstouch-taro", "meer-demo"),
        Repository("misstouch-taro", "private-notes", is_private=True),
    ]

    for repo in list_public_repos(samples):
        print(greet(repo))
